plugins/module_utils/dnac.py

```diff
@@ -405,13 +418,49 @@ def camel_to_snake_case(self, config):
         return new_config
 
 
-def log(msg, level='info', frameIncrement=0):
-    with open('dnac.log', 'a') as of:
+def log(msg, level='INFO', dnac_log_file_path='dnac.log', dnac_logs_append=True, frameIncrement=0):
+    """Logs/Appends messages into the specified log file or to dnac.log file by default
+
+    Args:
+        msg (str, required): The log message to be recorded.
+        level (str, optional): The log level, default is "info".
+                              The log level can be one of 'DEBUG', 'INFO', 'WARNING', 'ERROR', or 'CRITICAL'.
+        dnac_log_file_path (str, optional): The path to the log file, default is 'dnac.log'.
+        dnac_logs_append (bool, optional): A boolean indicating whether to append to an existing log file or create a new one, default is True.
+        frameIncrement (int, optional): The number of frames to increment in the call stack, default is 0.
+
+    Note:
+        Validates if the directory exists and raises an exception if it doesn't.
+        File gets created if it doesn't already exist.
+    """
+
+    global __first_log_written
+
+    # Validate dnac_log_file_path is absolute or relative
```

Comment:
All the below checks can be optimized. Can we rewrite? For every self.log call, we do all the below checks which are not required. As we use a global variable "_first_log_written" can we optimize it?

Reply:
I tried doing this with the logic you mentioned but when it comes to overwrite it overwrites every single line if done the way you specified at the end we are left with the log file with one line. 
Hence going with the way i had it before.